django/db/models/sql/compiler.py

```diff
@@ -350,6 +355,8 @@ def _order_by_pairs(self):
         # relatively expensive.
         if ordering and (select := self.select):
             for ordinal, (expr, _, alias) in enumerate(select, start=1):
+                if isinstance(expr, ColPairs):
+                    continue
```

Comment:
Hmm, no, it's possible to order by composite primary keys.

TBH, I'm not sure we need this. It has been such a long time ago I forgot why I added it. 😅 

I think I added it because `ColPairs` can't be aliased, so it doesn't make sense to add them to the `selected_exprs` dictionary.  But it doesn't really make a difference.